cassovary-core/src/main/scala/com/twitter/cassovary/graph/ArrayBasedDirectedGraph.scala

```diff
@@ -149,9 +153,8 @@ object ArrayBasedDirectedGraph {
           id = item.id
           newMaxId = newMaxId max item.maxId
           varNodeWithOutEdgesMaxId = varNodeWithOutEdgesMaxId max item.id
-          val edges = item.edges
-          edgesLength = edges.length
-          val newNode = ArrayBasedDirectedNode(id, edges, storedGraphDir)
+          edgesLength = item.edges.length
```

Comment:
we want to sort the edges if we are asked to (this was the original motivation). i.e., neighborsSorted is actually sortNeighbors. If neighbors is already sorted, well it is okay to sort it again if we are asked to. (If neighbors is already sorted and we do not want to sort them again, we will deal with the case when needed).